src/Ext/Techno/Hooks.Firing.cpp

```diff
@@ -794,3 +795,89 @@ DEFINE_HOOK(0x5223B3, InfantryClass_Approach_Target_DeployFireWeapon, 0x6)
 	R->EDI(pThis->Type->DeployFireWeapon == -1 ? pThis->SelectWeapon(pThis->Target) : pThis->Type->DeployFireWeapon);
 	return 0x5223B9;
 }
+
+DEFINE_HOOK(0x6FE562, TechnoClass_FireAt_BurstRandomTarget, 0x6)
+{
+	GET(TechnoClass*, pThis, ESI);
+	GET(WeaponTypeClass*, pWeapon, EBX);
+	GET(BulletClass*, pBullet, EAX);
+
+	if (!pBullet || pWeapon->Burst < 2)
+		return 0;
+
+	const auto pWeaponExt = WeaponTypeExt::ExtMap.Find(pWeapon);
+	if (!pWeaponExt || pWeaponExt->Burst_Retarget <= 0.0)
+		return 0;
+
+	int retargetProbability = pWeaponExt->Burst_Retarget > 1.0 ? 100 : (int)round(pWeaponExt->Burst_Retarget * 100);
```

Comment:
```suggestion
	int retargetProbability = min((int)round(pWeaponExt->Burst_Retarget * 100), 100);
```

Perhaps a min() function would be more readable.